code/game/machinery/machinery.dm

```diff
@@ -476,6 +478,31 @@ Class Procs:
 	if(.)
 		power_change()
 
+/obj/machinery/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/stack/nanopaste))
+		if(stat & BROKEN)
+			to_chat(user, "<span class='notice'>[src] is too damaged to be fixed with nanopaste!</span>")
+			return
+		if(obj_integrity == max_integrity)
+			to_chat(user, "<span class='notice'>[src] is fully intact.</span>")
+			return
+		if(being_repaired)
+			return
+		to_chat(user, "<span class='notice'>You start applying [O] to [src].</span>")
+		being_repaired = TRUE
+		var/result = do_after(user, 3 SECONDS, target = src)
+		being_repaired = FALSE
+		if(!result)
+			return
+		var/obj/item/stack/nanopaste/N = O
+		obj_integrity = clamp((obj_integrity + 50), obj_integrity, max_integrity)
```

Comment:
This needs to be placed after the `use`. Else you can have the situation you can't use it yet will heal

Reply:
```suggestion
		obj_integrity = min(obj_integrity + 50, max_integrity)
```
It can also be simplified since `obj_integrity + 50` can never be lower than `obj_integrity`